File: Java Labs/Patterns On Java/src/com/pattern/iterator/ClothingIterator.java
package com.pattern.iterator;

import java.util.List;

public class ClothingIterator implements Iterator {
    private List<ClothingItem> clothingItems;
    private int currentIndex;

    public ClothingIterator(List<ClothingItem> clothingItems) {
        this.clothingItems = clothingItems;
        this.currentIndex = 0;
    }

    @Override
    public boolean hasNext() {
        return currentIndex < clothingItems.size();
    }

    @Override
    public ClothingItem next() {
        if (hasNext()) {
            ClothingItem currentItem = clothingItems.get(currentIndex);
            currentIndex++;
            return currentItem;
        } else {
            return null;
        }
    }
}

File: Python Labs/com/pattern/iterator/ClothingIterator.py
from com.pattern.iterator.Iterator import Iterator


class ClothingIterator(Iterator):
    def __init__(self, clothing_items):
        self.clothing_items = clothing_items
        self.current_index = 0

    def has_next(self):
        return self.current_index < len(self.clothing_items)

    def next(self):
        if self.has_next():
            current_item = self.clothing_items[self.current_index]
            self.current_index += 1
            return current_item
        else:
            return None